t/semaphore.t

```diff
@@ -0,0 +1,1068 @@
+# vim:set ft= ts=4 sw=4 et fdm=marker:
+use lib 'lib';
+use Test::Nginx::Socket::Lua;
+use Cwd qw(cwd);
+
+#worker_connections(10140);
+master_process_enabled(1);
+workers(1);
+#log_level('warn');
+
+repeat_each(2);
+
+plan tests => repeat_each() * (blocks() * 3 + 1);
+
+my $pwd = cwd();
+
+no_long_string();
+#no_diff();
+
+#my $lua_default_lib = "/usr/local/openresty/lualib/?.lua";
+our $HttpConfig = <<_EOC_;
+    lua_package_path "$pwd/lib/?.lua;;";
+_EOC_
+
+our $HttpConfigInitByLua= <<_EOC_;
+    lua_package_path "$pwd/lib/?.lua;;";
+    init_by_lua '
+            local g = _G
+            local semaphore = require "ngx.semaphore"
+            local sem, err = semaphore.new(0)
+            g.err = err
+    ';
+_EOC_
+
+our $HttpConfigIntWorkerBy = <<_EOC_;
+    lua_package_path "$pwd/lib/?.lua;;";
+    init_worker_by_lua '
+            local semaphore = require "ngx.semaphore"
+            local sem = semaphore.new(0)
+            local ok, err = sem:wait(1)
+            local g = _G
+            g.test = err
+    ';
+
+_EOC_
+
+run_tests();
+
+__DATA__
```

Comment:
Sorry if I miss it, but have you tested *multiple* light threads *actually* waiting on the same semaphore? Most of the test cases just do one wait and one post.